Sloth.Web/Controllers/TransactionsController.cs

```diff
@@ -152,26 +157,180 @@ public async Task<IActionResult> Details(string id)
+        {
+            var transaction = await DbContext.Transactions
+                .Include(t => t.Scrubber)
+                .Include(t => t.JournalRequest)
+                .Include(t => t.Source)
+                    .ThenInclude(s => s.Team)
                 .Include(t => t.Transfers)
-                .Where(a => a.Id != transaction.Id && a.Source.Team.Slug == TeamSlug &&
-                    (
-                        a.KfsTrackingNumber == transaction.KfsTrackingNumber ||
-                        (a.ProcessorTrackingNumber != null && a.ProcessorTrackingNumber == transaction.ProcessorTrackingNumber) ||
-                        a.MerchantTrackingNumber == transaction.MerchantTrackingNumber
-                    )
-                ).ToListAsync();
-
-            var blobs = await DbContext.Blobs
-                .Where(b => b.TransactionBlobs.Select(tb => tb.TransactionId).Contains(transaction.Id)
-                    || b.Scrubbers.SelectMany( s => s.Transactions.Select(t => t.Id)).Contains(transaction.Id))
-                .ToListAsync();
+                .Include(t => t.ReversalTransaction)
+                .Include(t => t.ReversalOfTransaction)
+                .Include(t => t.StatusEvents)
+                .AsNoTracking()
+                .FirstOrDefaultAsync(t => t.Id == id && t.Source.Team.Slug == TeamSlug);
+
+            if (transaction == null)
+            {
+                ErrorMessage = "Transaction not found";
+                return RedirectToAction(nameof(Index));
+            }
+
+            if (transaction.Status != TransactionStatuses.Rejected && !transaction.IsStale())
+            {
+                ErrorMessage = "Transaction is not Stale (Processing for more than 5 days) or Rejected";
+                return RedirectToAction(nameof(Details), new { id });
+            }
+
+            foreach (var transfer in transaction.Transfers)
+            {
+                transfer.AccountingDate = transfer.AccountingDate?.ToPacificTime();
+            }
+
+            var model = await GetTransactionDetailsViewModel(transaction);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [Authorize(Policy = PolicyCodes.TeamManager)]
+        public async Task<IActionResult> Edit(string id, TransactionEditViewModel transaction)
+        {
+            if (transaction == null || string.IsNullOrWhiteSpace(id))
+            {
+                ErrorMessage = "No transaction specified";
+                return RedirectToAction(nameof(Index));
+            }
+
+            var currentTransaction = await DbContext.Transactions
+                .Include(t => t.Scrubber)
+                .Include(t => t.JournalRequest)
+                .Include(t => t.Source)
+                    .ThenInclude(s => s.Team)
+                .Include(t => t.Transfers)
+                .Include(t => t.ReversalTransaction)
+                .Include(t => t.ReversalOfTransaction)
+                .Include(t => t.StatusEvents)
```

Comment:
why are all these includes needed?

Reply:
The edit form is pretty close in format to the Details form. I thought it would be helpful to keep as much context as possible when someone is editing a transaction.